### ISSUE TITLE
This project violates copyright

### ISSUE THREAD

**Issue by VitaliyMF** [henkmollema/WkWrap#2]
@henkmollema I ask you to remove this repository and nuget package from nuget.org because the code of the wkhtmltopdf wrapper is based on commercial component [NReco.PdfGenerator](https://www.nrecosite.com/pdf_generator_net.aspx). You slightly modified the code, but this fact can be easily proven by names / code snippets, you may contact me if you want to know what exactly I mean.

If you will not react on this request in several days I'll open a dispute with github.com/nuget.org administration.

**Comment by henkmollema** [comment]
I "inherited" this project as @nomoreload removed his original repository and I had a fork of it (perhaps because of this?). Figured I'll continue it. 

I do wonder how similar it is though. This tool is merely wrapper around the `Process` class. It does not provide the `wkhtmltopdf` execute (which NReco.PdfGenerator does). The project structure has changed and more functionality (such as .NET Core support and async methods) have been added. NReco.PdfGenerator hasn't been updated for more than a year and does not target .NET Standard.

I'll be interested in the "names / code snippets" which hare similar.

**Comment by VitaliyMF** [comment]
> I "inherited" this project as @nomoreload <https://github.com/nomoreload> removed
> his original repository and I had a fork of it (perhaps because of this?).
> Figured I'll continue it.


Roman Bukin @nomoreload removed the repository because of my request.

Here is a copy of our communication:

---------- Forwarded message ----------
> From: Роман Букин <nomoreload@gmail.com>
> Date: Wed, May 3, 2017 at 4:05 PM
> Subject: Re: Regarding code snippets in your open source project WkWrap
> To: NReco Components <info@nrecosite.com>
>
>
> I've removed it.
>
> 2017-05-03 13:16 GMT+03:00 NReco Components <info@nrecosite.com>:
>
>> Hi Roman,
>>
>> I've found your open source wkhtmltopdf wrapper (WkWrap
>> <https://github.com/NoMoreLoad/WkWrap>) and it some code snippets are
>> very similar to our commercial component NReco.PdfGenerator
>> <https://www.nuget.org/packages/NReco.PdfGenerator>:
>>
>>    - class name is the same: HtmlToPdfConverter
>>    - public event EventHandler<DataReceivedEventArgs> LogReceived  --
>>    exactly the same
>>    - CheckWkHtmlProcess method is EXACTLY the same
>>    - CheckExitCode method is EXACTLY like in our component
>>    - and the last, special processing of wkhtmltopdf errors that
>>    contains EXACTLY the same constants as in our component:
>>
>> private static readonly string[] IgnoredWkHtmlToPdfErrors =
>> {
>> "Exit with code 1 due to network error: HostNotFoundError",
>> "Exit with code 1 due to network error: ContentNotFoundError",
>> "Exit with code 1 due to network error: ContentOperationNotPermittedEr
>> ror",
>> "Exit with code 1 due to network error: ProtocolUnknownError",
>> "Exit with code 1 due to network error: UnknownContentError",
>> "QFont::setPixelSize: Pixel size <= 0"
>> };
>>
>>
>> I may assume that you've disassembled NReco.PdfGenerator.dll or purchased
>> source code package; in any way license doesn't allow you to use the code
>> in this way.
>> You need to remove your project from github as its git history already
>> contains copyrighted code. You may change your component source, remove any
>> similarities with copyrighted code and then publish it on github again.
>>
>> *If you will ignore this request in 5 days period we'll contact github
>> support with copyright abuse request according to github terms of
>> service: https://help.github.com/articles/github-terms-of-service/#e-copyright-infringement-and-dmca-policy
>> <https://help.github.com/articles/github-terms-of-service/#e-copyright-infringement-and-dmca-policy>*
>>
>> Regards,
>> NReco Team
>> http://www.nrecosite.com
>>
>
*If you forked from this code you violate copyright of my component too.*

I do wonder how similar it is though. This tool is merely wrapper around the
>  Process class. It does not provide the wkhtmltopdf execute (which
> NReco.PdfGenerator does). The project structure has changed and more
> functionality (such as .NET Core support and async methods) have been
> added. NReco.PdfGenerator hasn't been updated for more than a year and does
> not target .NET Standard.


You're wrong because PdfGenerator supports .NET Core with
NReco.PdfGenerator.LT
<https://www.nuget.org/packages/NReco.PdfGenerator.LT/> nuget package
(however, it is not free and requires a license key).

Regards,
Vitaliy Fedorchenko
https://www.nrecosite.com

**Comment by VitaliyMF** [comment]
Henk,

I understand that you spent some time on adding modifications, reorganized
the code etc but this cannot change the fact that you started from
repository that originally violates the copyright, and nothing can change
this fact (original WkWrap author makes very bad thing when he decided to
decompile my component and make his own 'like this').

I ask you to remove github repo and delete nuget package on nuget.org. You
can create new component similar to PdfGenerator from scratch if you like,
but you cannot use code of WkWrap.

BTW if you need a replacement for WkWrap in your projects I can offer
significant discount for PdfGenerator.

BR,
Vitaliy Fedorchenko
https://www.nrecosite.com

**Comment by henkmollema** [comment]
I agree that decompiling assemblies and copy-pasting the code is a very bad thing to do. It was never my intention to continue on that. If I had known I wouldn't have republished the package as-is. I will take it offline and might come back with my own version of it.

> BTW if you need a replacement for WkWrap in your projects I can offer significant discount for PdfGenerator.

Sorry, but no thanks. Honestly, I don't know what to think about this comment. It makes your point lose credibility.
